src/main/java/launutsapi/LaunutsController.java

```diff
@@ -1,8 +1,155 @@
 package launutsapi;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.json.simple.JSONArray;
+import org.json.simple.parser.ParseException;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
 
 @RestController
 public class LaunutsController {
+	
+	@Autowired
+	private NutsService nut_service;
+	
+	@Autowired
+	private LauService lau_service;
+	
+	private Nuts nut;
+	private Lau lau;
+	
+	private String notification = "Please enter query in correct format."
+		     + "Some correct formats are like these "
+		     + "http://localhost:8080/launuts/nuts/DEA47/json, "
+		     + "http://localhost:8080/launuts/nuts/DEA47/ttl, "
+		     + "http://localhost:8080/launuts/nuts/paderborn/json, "
+		     + "http://localhost:8080/launuts/nuts/paderborn/ttl, "
+		     + "http://localhost:8080/launuts/nuts/ttl, "
+		     + "http://localhost:8080/launuts/nuts/json ";
+	
+	@RequestMapping("/launuts/nuts/{query_string}/json")
+	public List<Nuts> getNutsJson(@PathVariable String query_string) throws IOException, ParseException {
+		
+		//Validate input: If user trying to enter a nut id
+		if(query_string.matches(".*\\d.*") && !(query_string.toLowerCase().matches("^((de)|(de\\D))((\\d{1})|(\\d{2})|(\\d{3}))$"))) {
+	
+			nut = new Nuts(notification);	
+			return Arrays.asList(nut);	
+		}
+		else 
+			return Arrays.asList(nut_service.getNutsJson(query_string));	
+			
+	}
+	
+	@RequestMapping("/launuts/lau/{query_string}/json")
+	public List<Lau> getLauJson(@PathVariable String query_string) throws IOException, ParseException {
+		
+		//Validate input: If user trying to enter a nut id
+		if(query_string.matches(".*\\d.*") && !(query_string.toLowerCase().matches("^(de_)\\d{8}"))) {
+	
+			lau = new Lau(notification);	
+			return Arrays.asList(lau);	
+		}
+		else 
+			return Arrays.asList(lau_service.getLauJson(query_string));	
+			
+	}
+	
+	
+	@RequestMapping("/launuts/nuts/{query_string}/ttl")
```

Comment:
This rest api is not generating the expected result. It displays: wrong format to the correct path specified.
I have also attached image below for reference.
![error](https://user-images.githubusercontent.com/52550720/84090498-4bf8d580-a9f2-11ea-991e-d8b72f25f6c1.JPG)

Reply:
The regex for this query has been fixed. Thanks for the review.

Reply:
It works fine now.